app/src/main/java/ch/epfl/sdp/drone3d/ui/weather/WeatherInfoActivity.kt

```diff
@@ -56,7 +57,7 @@ class WeatherInfoActivity : AppCompatActivity() {
         nonExistentMission = findViewById(R.id.nonExistentMission)
 
         @Suppress("UNCHECKED_CAST")
-        location = intent.getSerializableExtra(MissionViewAdapter.MISSION_PATH) as ArrayList<LatLng>?
+        location = intent.getSerializableExtra(ItineraryShowActivity.FLIGHTPATH_INTENT_PATH) as ArrayList<LatLng>?
```

Comment:
This can be replaced with 
```suggestion
        location = intent.getParcelableArrayListExtra(ItineraryShowActivity.FLIGHTPATH_INTENT_PATH)
```
which removes the need of the Supress